test/chainnotifier_mock.go

```diff
@@ -112,26 +126,35 @@ func (c *mockChainNotifier) RegisterConfirmationsNtfn(ctx context.Context,
 
 		select {
 		case m := <-c.lnd.ConfChannel:
-			if bytes.Equal(m.Tx.TxOut[0].PkScript, pkScript) {
-				select {
-				case confChan <- m:
-				case <-ctx.Done():
+			c.Lock()
+			for i := 0; i < len(c.confRegistrations); i++ {
+				r := c.confRegistrations[i]
+
+				// Whichever conf notifier catches the confirmation
+				// will forward it to all matching subscibers.
+				if bytes.Equal(m.Tx.TxOut[0].PkScript, r.PkScript) {
+					// Unregister the "notifier".
+					c.confRegistrations = append(
+						c.confRegistrations[:i], c.confRegistrations[i+1:]...,
```

Comment:
Isn't `i+1` a crash if it is the last reg? Probably not as I see similar examples on internet

Reply:
it's valid (https://play.golang.org/p/TXP0B4ZtCNw)